src/Phinx/Db/Adapter/SqlServerAdapter.php

```diff
@@ -1109,17 +1103,23 @@ public function getSqlType($type, $limit = null)
      * @internal param string $sqlType SQL type
      *
      * @param string $sqlType SQL Type definition
+     * @param int|null $char_length Column length value
      *
      * @throws \Phinx\Db\Adapter\UnsupportedColumnTypeException
      *
      * @return string Phinx type
      */
-    public function getPhinxType($sqlType)
+    public function getPhinxType($sqlType, $char_length = null)
     {
         switch ($sqlType) {
             case 'nvarchar':
             case 'varchar':
-                return static::PHINX_TYPE_STRING;
+                if ($char_length == -1) {
+                    // max char length
+                    return static::PHINX_TYPE_TEXT;
+                } else {
```

Comment:
the else is not necessary with early returns.

Reply:
I went ahead and removed that.